File: src/main/java/com/dingdangmaoup/resin/pura/ui/ShowResinConfDialog.java
package com.dingdangmaoup.resin.pura.ui;

import com.dingdangmaoup.resin.pura.ResinBundle;
import java.awt.Dimension;
import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;
import javax.swing.JButton;
import javax.swing.JDialog;
import javax.swing.JPanel;
import javax.swing.JTextArea;

public class ShowResinConfDialog extends JDialog {
  private JPanel contentPane;
  private JButton buttonOK;
  private JTextArea textArea;

  public ShowResinConfDialog(String text) {
    setContentPane(contentPane);
    setModal(true);
    getRootPane().setDefaultButton(buttonOK);

    buttonOK.addActionListener(new ActionListener() {
      @Override
      public void actionPerformed(ActionEvent e) {
        onOK();
      }
    });

    setTitle(ResinBundle.message("message.text.resin.conf.altered"));

    textArea.setText(text);
    textArea.setEditable(false);
    contentPane.setPreferredSize(new Dimension(640, 480));
  }

  private void onOK() {
    // add your code here
    dispose();
  }
}

File: src/main/kotlin/com/dingdangmaoup/resin/pura/ui/ShowResinConfDialog.kt
package com.dingdangmaoup.resin.pura.ui

import com.dingdangmaoup.resin.pura.ResinBundle
import java.awt.Dimension
import com.intellij.openapi.ui.DialogWrapper
import com.intellij.ui.components.JBScrollPane
import com.intellij.ui.dsl.builder.AlignX
import com.intellij.ui.dsl.builder.panel
import javax.swing.Action
import javax.swing.JComponent
import javax.swing.JTextArea

class ShowResinConfDialog(text: String) : DialogWrapper(true) {
    private val textArea = JTextArea().apply {
        this.text = text
        isEditable = false
    }
    private val contentPanel = panel {
        row {
            cell(JBScrollPane(textArea)).align(AlignX.FILL)
        }.resizableRow()
    }

    init {
        title = ResinBundle.message("message.text.resin.conf.altered")
        init()
        contentPanel.preferredSize = Dimension(640, 480)
    }

    override fun createCenterPanel(): JComponent = contentPanel

    override fun createActions(): Array<Action> = arrayOf(okAction)
}
